TaskHero/TaskDetailViewController.swift

```diff
@@ -11,63 +11,119 @@ import UIKit
         }
     }
     
+    func setPercentBarAndLabel() {
+        let percentComplete = round((taskInstance?.getPercentComplete())! * 100)
+        percentLabel.text = "\(percentComplete)%"
+        progressBarContainerView.layer.borderWidth = 2
+        progressBarContainerView.layer.borderColor = UIColor.black.cgColor
+        let viewMaxWidth = progressBarContainerView.frame.width
+        progressBarTrailingConstraint.constant = (viewMaxWidth - (CGFloat(percentComplete) / 100 * viewMaxWidth))
+    }
+    
     @IBAction func onBackButton(_ sender: Any) {
         _ = navigationController?.popViewController(animated: true)
-    }    
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if let vc = segue.destination as? StepDetailViewController {
+            vc.step = steps![selectedCell!]
+        } else if let vc = segue.destination as? ChatViewController {
+            vc.task = taskInstance
+        }
+    }
     
 }
 
 extension TaskDetailViewController: UITableViewDelegate, UITableViewDataSource {
+    
+    func loadTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.rowHeight = UITableViewAutomaticDimension
+        tableView.estimatedRowHeight = 80
+        tableView.register(UINib(nibName: stepCellIdentifier, bundle: nil), forCellReuseIdentifier: stepCellIdentifier)
+    }
+    
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        currentSelectedCellRowNum = indexPath.row
-        performSegue(withIdentifier: "TaskDetailToStepDetail", sender: self) // send a task which selected
+        selectedCell = indexPath.row
+        tableView.deselectRow(at: indexPath, animated: true)
+        performSegue(withIdentifier: "TaskDetailToStepDetail", sender: self)
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: stepCellIdentifier, for: indexPath) as! StepCell
-        cell.step = steps?[indexPath.row]
-
+        let cell = tableView.dequeueReusableCell(withIdentifier: stepCellIdentifier, for: indexPath) as! StepDetailCell
+        cell.step = steps?[indexPath.row]        
+        if taskInstance?.getNextStep() == steps?[indexPath.row] {
+            cell.nextStep = true
+        }
+        cell.loadCell()
+        cell.delegate = self
         return cell
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         return steps?.count ?? 0
     }
+    
+}
+
+extension TaskDetailViewController: TaskInstanceUpdateDelegate {
+    
+    func taskInstanceUpdated() {
+        ParseClient.sharedInstance.updateTaskInstance(taskInstance: taskInstance!, success: { () -> () in
+            self.tableView.reloadData()
+            self.setPercentBarAndLabel()
```

Comment:
TODO: send a push to the assignees of the next step letting them know the last step was marked "complete". Or if the task is complete, send a push to all users letting them know.